src/ODE.jl

```diff
@@ -18,6 +18,41 @@ export ode4s, ode4ms, ode4
 #    oderosenbrock, ode4s, ode4s_kr, ode4s_s,
 #    ode4ms, ode_ms
 
+###############################################################################
+## HELPER FUNCTIONS
+###############################################################################
+
+# estimator for initial step based on book
+# "Solving Ordinary Differential Equations I" by Hairer et al., p.169
+function hinit(F, x0, t0, p, reltol, abstol)
+    tau = max(reltol*norm(x0, Inf), abstol)
+    d0 = norm(x0, Inf)/tau
+    f0 = F(t0, x0)
+    d1 = norm(f0, Inf)/tau
+    if d0 < 1e-5 || d1 < 1e-5
+        h0 = 1e-6
+	else
+        h0 = 1e-2d0/d1
+	end
+    # perform Euler step
+    x1 = x0 + h0*f0
+    f1 = F(t0 + h0, x1)
+    # estimate second derivative
+    d2 = norm(f1 - f0, Inf)/(tau*h0)
+    if max(d1, d2) <= 1e-15
+        h1 = max(1e-6, 1e-3*h0)
+	else
+        pow = -(2. + log10(max(d1, d2)))/(p + 1.)
+        h1 = 10.^pow
+	end
+	h = min(100.0*h0, h1)
```

Comment:
This should also not be indented, to make it clerarer that `h` gets this value regardless of whether `max(d1,d2) <= 1e-15` or not.